### PR TITLE
Expose scripts to create deployments of deku networks

### PR DESCRIPTION
**Pull request by ulrikstrid**
## Problem

The infra team wants to have a more reproducible way of deploying deku nodes

## Solution

Expose some scripts to generate contracts that we can use to deploy deku nodes

### CODE REVIEW

**Comment by ulrikstrid** [comment]
@d4hines I think this is reviewable. It still includes some stuff that is specific to our environment but maybe that can slide until we are going to deploy the mainnet where we'll also need to generalize it?

File: nix/scripts/create-deku-deployment.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+VALIDATORS=( $(seq 0 "$((NUMBER_OF_NODES - 1))") )
+
+# base name to use
+KEY_NAME=${2:-"deku-testnet"}
+
+# tezos RPC node
+RPC_NODE=${3:-"https://ithacanet.tezos.marigold.dev"}
+
+# contract names
+CONSENSUS_NAME="$KEY_NAME-discovery"
```
Comment (gsebil08) [inline nix/scripts/create-deku-deployment.sh:50]: CONSENSUS_NAME=DISCOVERY_NAME
is it normal?

File: nix/scripts/create-deku-deployment.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+print_success "Consensus discovery deployed"
+
+print_info "Getting contract addresses"
+# Get contract data
+TEZOS_CONSENSUS_ADDRESS="$(tezos-client --endpoint "$RPC_NODE" show known contract "$CONSENSUS_NAME" | grep KT1 | tr -d '\r')"
+TEZOS_DISCOVERY_ADDRESS="$(tezos-client --endpoint "$RPC_NODE" show known contract "$DISCOVERY_NAME" | grep KT1 | tr -d '\r')"
+print_success "Got contract addresses"
+
+print_info "Generating tezos.json for nodes"
+# generate tezos.json
```
Comment (gsebil08) [inline nix/scripts/create-deku-deployment.sh:129]: I guess we can get rid of this comment

**Review by gsebil08 — COMMENTED** [review]
Is it possible to try it on local?

File: nix/scripts/create-consensus-storage.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#! /usr/bin/env bash
+
+set -e
+
+consensus_storage=$(
+  # Step 3: After having the Deku identities, we will configure and deploy
```
[the remark this answers is not in the document]
Comment (ulrikstrid) [reply nix/scripts/create-consensus-storage.sh:6]: Nope, it should be changed/removed. Thank you

File: nix/scripts/create-discovery-storage.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#! /usr/bin/env bash
+
+set -e
+
+  ## most of the noise in this function is because of indentation
```
[the remark this answers is not in the document]
Comment (ulrikstrid) [reply nix/scripts/create-discovery-storage.sh:5]: This is borrowed from sandbox.sh

**Comment by d4hines** [comment]
Closing this in favor #708 and friends

**Pull request closed without merging**